Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -669,13 +669,186 @@ end Integrable
 
 open MeasureTheory
 
+/-!
+### Oscillation
+-/
+
+section oscillation
+
+open EMetric
+
+/-- The oscillation of `f` at `x`. -/
+noncomputable def oscillation (f : ℝⁿ → E) (x : ℝⁿ) : ENNReal :=
+  ⨅ S ∈ (𝓝 x).map f, EMetric.diam S
+
+/-- The oscillation of `f` at `x` is 0 whenever `f` is continuous at `x`. -/
+theorem oscillation_zero_of_continuousAt {f : ℝⁿ → E} {x : ℝⁿ} (hf : ContinuousAt f x) :
+    oscillation f x = 0 := by
+  refine le_antisymm (ENNReal.le_of_forall_pos_le_add fun ε hε _ ↦ ?_) (zero_le _)
+  rw [zero_add]
+  have : EMetric.ball (f x) (ε / 2) ∈ (𝓝 x).map f :=
+    hf <| EMetric.ball_mem_nhds _ (by simp [ne_of_gt hε])
+  refine (biInf_le EMetric.diam this).trans (le_of_le_of_eq diam_ball ?_)
+  refine (ENNReal.mul_div_cancel' ?_ ?_) <;> norm_num
```

Comment:
```suggestion
  exact (ENNReal.mul_div_cancel' ?_ ?_) <;> norm_num
```

Reply:
I don't think the suggested change works. It could be changed to 

`exact ENNReal.mul_div_cancel' (by norm_num) (by norm_num)`

if that's considered preferable.

Reply:
Yes, that's what I meant. Sorry about that.